src/output.ts

```diff
@@ -30,14 +29,17 @@ export type ActionInContext = {
   committed?: boolean;
 }
 
-export class TerminalOutput {
+export class Output {
   private _currentAction: ActionInContext | undefined;
   private _lastAction: ActionInContext | undefined;
   private _lastActionText: string | undefined;
-  private _out: Writable;
+  private _outputs: OutputMultiplexer;
 
-  constructor(browserName: string, launchOptions: playwright.LaunchOptions, contextOptions: playwright.BrowserContextOptions, out: Writable, deviceName: string | undefined) {
-    this._out = out;
+  constructor(browserName: string, launchOptions: playwright.LaunchOptions, contextOptions: playwright.BrowserContextOptions, out: Writable, outputFile: string | undefined, deviceName: string | undefined) {
+    this._outputs = new OutputMultiplexer()
+    this._outputs.add(new TerminalOutput(out))
```

Comment:
Steps to make it better design-wise, based on the principle of least knowledge:
- Rename 'Output' to 'Generator'
- Declare 'GeneratorOutput' beside it, as far as Generator is concerned it emits into the GeneratorOutput.
- Instantiate Multiplexer externally to the Generator. You can already see how `out` and `outputFile` are only used to create it. Generator should not know about multiplexing.
- Use `this._output`, single, it is just an output sink